### PR TITLE
set_settings method to fill UI widgets

### PR DESCRIPTION
**Pull request by ghtmtt**
(no merge)

### CODE REVIEW

**Comment by ghtmtt** [comment]
@nyalldawson not finished yet. Daadefined buttons are commented (`setProperty` method still to study), and added something in the tests. Let me know what you think.

**Comment by ghtmtt** [comment]
> This is looking good - exactly on the right track!

waiting for your review suggestion. Should I prepare a layer to add to the test suite? Tests are also failing because `self.layer.currentLayer().id()` is `None`. With a fake layer to play with we can enhance the tests. I also think it might me useful in the future. Or is it better to create every time a temp layer directly in the test files?

**Comment by nyalldawson** [comment]
> Should I prepare a layer to add to the test suite? 

That's a good idea - we'll need this elsewhere too. 

> Tests are also failing because self.layer.currentLayer().id() is None. With a fake layer to play with we can enhance the tests. I also think it might me useful in the future.

Agreed, a layer is needed. But we will likely also need to handle cases when there's no vector layers in a project (e.g. if the plugin is loaded but only raster layers are available). So it would be good to fix the test in this situation too...

File: DataPlotly/data_plotly_dialog.py [hunk 1]
```diff
@@ -984,8 +1031,85 @@ def get_settings(self) -> PlotSettings:
...
+
+        settings = self.get_settings()
+
+        # Set the plot properties
+        self.layer_combo.setLayer(settings.properties['layer_id'])
+        self.selected_feature_check.setChecked(settings.properties['features_selected'])
+        self.x_combo.setExpression(settings.properties['x_combo'])
+        self.y_combo.setExpression(settings.properties['y_combo'])
+        self.z_combo.setExpression(settings.properties['z_combo'])
+        self.in_color_combo.setColor(settings.properties['in_color_combo'])
```
Comment (nyalldawson) [inline DataPlotly/data_plotly_dialog.py]: mental note: we'll need to convert colors to strings for storage. leave that one with me.

**Comment by ghtmtt** [comment]
> Right, but now this will need to be tweaked to something like

but this will always lead to an error, right? in the dialog we don't have any loaded layers and `self.layer_combo.currentLayer().id()` throws the error `AttributeError: 'NoneType' object has no attribute 'id'`.

**Comment by nyalldawson** [comment]
That will need to be caught

    self.layer_combo.currentLayer().id() if self.layer_combo.currentLayer() else None

**Comment by ghtmtt** [comment]
do you have any idea why it is still failing? it seems related to another method in the testing framework..

**Comment by nyalldawson** [comment]
@ghtmtt 

The tests are passing, it's just the lint which is complaining:

image

**Comment by nyalldawson** [comment]
You'll need to add

    # pylint: disable=R0915

after the definition for this function (search elsewhere in that module for how to add this), in this case there's no issue having a lot of lines in this particular function.

**Comment by ghtmtt** [comment]
finally :) can I merge this?

**Comment by nyalldawson** [comment]
sure, go ahead!

**Pull request merged**